packages/core/admin/admin/src/content-manager/components/RelationInputDataManager/RelationInputDataManager.js

```diff
@@ -0,0 +1,256 @@
+import PropTypes from 'prop-types';
+import React, { memo, useEffect, useMemo } from 'react';
+import { useIntl } from 'react-intl';
+import get from 'lodash/get';
+
+import { useCMEditViewDataManager, NotAllowedInput, useQueryParams } from '@strapi/helper-plugin';
+
+import { RelationInput } from '../RelationInput';
+import { useRelation } from '../../hooks/useRelation';
+import { connect, select, normalizeRelations } from './utils';
+import { PUBLICATION_STATES, RELATIONS_TO_DISPLAY, SEARCH_RESULTS_TO_DISPLAY } from './constants';
+import { getTrad } from '../../utils';
+
+export const RelationInputDataManger = ({
+  editable,
+  description,
+  intlLabel,
+  isCreatingEntry,
+  isFieldAllowed,
+  isFieldReadable,
+  labelAction,
+  mainField,
+  name,
+  queryInfos: { endpoints, defaultParams, shouldDisplayRelationLink },
+  placeholder,
+  required,
+  relationType,
+  size,
+  targetModel,
+}) => {
+  const { formatMessage } = useIntl();
+  const { connectRelation, disconnectRelation, loadRelation, modifiedData, slug, initialData } =
+    useCMEditViewDataManager();
+  const [{ query }] = useQueryParams();
+
+  const { relations, search, searchFor } = useRelation(`${slug}-${name}-${initialData?.id ?? ''}`, {
+    relation: {
+      enabled: get(initialData, name)?.count !== 0 && !!endpoints.relation,
+      endpoint: endpoints.relation,
+      pageParams: {
+        ...defaultParams,
+        locale: query?.plugins?.i18n?.locale,
```

Comment:
I understand the need here but it goes against the entire idea of having a plugin ecosystem if we start hardcoding specific behaviors depending on whether or not a plugin is installed...

Honestly, I believe that if we start doing such statement in the CM codebase for a plugin it means that we need to move this plugin into the core and remove the plugin notion for i18n. If you start doing this for one plugin how can you make sure you won't do it again for another one ? In the end you might end up maintaining a huge tree of statement with conditions based on whether or not a plugin is installed.

Reply:
This is why we introduced the queryInfos in the layout to let plugins define search params without having to handle conditions in the CM.

Reply:
You are right: we were a bit in a rush and had to put this in as a temporary solution. I will move this out again and pass down the current locale. Since `queryInfos` only holds i18n information from now on (we can not generate the name of the fetch route from the content-type and need the entity id too), I'd like to rename it, to reflect that.

Reply:
Working on a fix here: https://github.com/strapi/strapi/pull/14555